jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -782,15 +782,14 @@ private void checkSegment(String uri, int segment, int end, boolean param)
 
         // Look for segment in the ambiguous segment index.
         Boolean ambiguous = __ambiguousSegments.get(uri, segment, end - segment);
-        if (ambiguous == Boolean.TRUE)
+        if (ambiguous != null)
```

Comment:
Do we really want the `PARAM` check to be inside this block?
Then paths like `/path/test;param/info` are not ambiguous with `Violation.PARAM` even though they are decoded to `/path/test/info`. 

I can see tests like `{";/bar", "/bar", EnumSet.of(Violation.EMPTY)},` but not sure why we wouldn't classify this as ambiguous.

Reply:
I don't understand?   `/path/test;param/info` is not ambiguous as it decodes to exactly `/path/test/info`.   Paths like `/path/..;param/info` and `/path/%2e%2e;param/info` are ambiguous because the RFC strictly says  they should not be normalized, but once params are stripped they could be.